### PR TITLE
feat: add CI to publish to cargo

### PR DESCRIPTION
**Pull request by obycode**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Install dependencies
+        if: steps.cache-cargo.outputs.cache-hit != 'true'
+        run: cargo install grcov
+
+      - name: Unit Tests
+        env:
+          RUSTFLAGS: "-Zinstrument-coverage"
+          LLVM_PROFILE_FILE: "clarity-repl-%p-%m.profraw"
```
Comment (CharlieC3) [inline .github/workflows/ci.yaml]:
```suggestion
          LLVM_PROFILE_FILE: "dap-types-%p-%m.profraw"
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+
+      - name: Build - Cargo
+        run: cargo build --release
+
+      - name: Unit Tests - Cargo
+        run: cargo test --release
+
+      - name: Publish clarity-repl to crates.io
```
Comment (CharlieC3) [inline .github/workflows/ci.yaml]:
```suggestion
      - name: Publish dap-types to crates.io
```

**Review by CharlieC3 — CHANGES_REQUESTED** [review]
Just a couple minor changes.

**Comment by CharlieC3** [comment]
@obycode Forgot to mention the package.json was needed for Semantic Release to work. I added one and enabled Husky for commit message sanitizing.

**Comment by obycode** [comment]
> @obycode Forgot to mention the package.json was needed for Semantic Release to work. I added one and enabled Husky for commit message sanitizing.

Excellent. Thanks!

**Review by CharlieC3 — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**